### PR TITLE
feat: support `SHOW TABLE STATUS`

### LINKED ISSUE #1233: feat(sql): support `SHOW TABLE STATUS` statement
> `SHOW TABLE STATUS` provides more extensive information about each non-TEMPORARY table

Here it may quite different with other DMBS's e.g [mariadb's show table status](https://mariadb.com/kb/en/show-table-status/) yet. But we'll first support show infos mostly required by monitoring

Fileds include:
- db: database name
- name: table name
- storage_mode: memory/disk
- rows: number of rows in the table
- partition_num: total partition number
- partition_unalive_num: unalive partition number
- replica_num: replica number
- data_length: table data + index size (in bytes) in memory 
- disk_size: disk file size (in memory)

## dependency
- #1260 
- #1259

### PR DESCRIPTION
**Pull request by aceforeverd**
close #1233 

## SQL Rules
```
SHOW TABLE STATUS
```

- `FROM|IN` clause, `LIKE` clause is not supported
- if no database is selected with `use db`, all tables' status will shown, except tables located in hidden databases
- if a database is selected, only tables inside that database will shown

## Other minor tweaks

- change the output schema of `SHOW Components`. For schema, the column name default to Upper case first char then lower case all following chars

### CODE REVIEW

**Comment by aceforeverd** [comment]
[online monitoring](https://trello.com/c/arsSP4R6/1-online-monitoring)

File: src/cmd/sql_cmd_test.cc [hunk 1]
```diff
@@ -42,6 +45,67 @@ inline std::string GenRand() {
...
+            if (expect[idx][i].expect_.has_value()) {
+                EXPECT_EQ(expect[idx][i].expect_.value(), val) << "expect[" << idx << "][" << i << "]";
+            }
+            if (expect[idx][i].expect_not_.has_value()) {
+                EXPECT_NE(expect[idx][i].expect_not_.value(), val) << "expect_not[" << idx << "][" << i << "]";
+            }
+        }
+    }
+
+    CellExpectInfo i = "abc";
```
Comment (aceforeverd) [inline src/cmd/sql_cmd_test.cc]: rm those

File: src/cmd/sql_cmd_test.cc [hunk 2]
```diff
@@ -372,6 +420,62 @@ TEST_P(DBSDKTest, ShowComponents) {
...
+    // default schedule interval is 2s
+    absl::SleepFor(absl::Seconds(10));
+
+    // test
+    hybridse::sdk::Status status;
+    auto rs = sr->ExecuteSQL("show table status", &status);
+    ASSERT_EQ(status.code, 0);
+    ExpectResultSetStrEq({{"Table_id", "Table_name", "Database_name", "Storage_type", "Rows", "Memory_data_size",
+                           "Disk_data_size", "Partition", "Partition_unalive", "Replica"},
+                          {{}, tb_name, db_name, "memory", "1", {{}, "0"}, {}, "1", "0", "1"}},
```
Comment (aceforeverd) [inline src/cmd/sql_cmd_test.cc]: 1. when is disk data size not empty
2. size unit ?

File: src/sdk/sql_cluster_router.cc [hunk 3]
```diff
@@ -3231,5 +3229,70 @@ std::vector<::hybridse::vm::AggrTableInfo> SQLClusterRouter::GetAggrTables() {
...
 }
 
+static const std::initializer_list<std::string> GetTableStatusSchema() {
+    static const std::initializer_list<std::string> schema = {
+        "Table_id",         "Table_name",     "Database_name", "Storage_type",      "Rows",
+        "Memory_data_size", "Disk_data_size", "Partition",     "Partition_unalive", "Replica"};
+    return schema;
+}
+
+// output schema:
```
Comment (aceforeverd) [inline src/sdk/sql_cluster_router.cc:3240]: comment about the meanings

File: src/sdk/sql_cluster_router.cc [hunk 4]
```diff
@@ -3231,5 +3229,70 @@ std::vector<::hybridse::vm::AggrTableInfo> SQLClusterRouter::GetAggrTables() {
...
+        auto table_name = tinfo.name();
+        auto db = tinfo.db();
+        // TODO(aceforeverd): support disk type
+        std::string storage_type = "memory";
+        auto partition_num = tinfo.partition_num();
+        auto replica_num = tinfo.replica_num();
+        uint64_t rows = 0, mem_bytes = 0, disk_bytes = 0;
+        uint32_t partition_unalive = 0;
+        for (auto& partition_info : tinfo.table_partition()) {
+            // rows += partition_info.record_cnt();
```
Comment (aceforeverd) [inline src/sdk/sql_cluster_router.cc]: recheck

File: src/sdk/sql_cluster_router.cc [hunk 5]
```diff
@@ -3231,5 +3229,70 @@ std::vector<::hybridse::vm::AggrTableInfo> SQLClusterRouter::GetAggrTables() {
...
+    cluster_sdk_->GetNsClient()->ShowTable("", "", true, tables, msg);
+
+    std::vector<std::vector<std::string>> data;
+    data.reserve(tables.size());
+
+    std::for_each(tables.cbegin(), tables.cend(), [&data](const nameserver::TableInfo& tinfo) {
+        if (nameserver::IsHiddenDb(tinfo.db())) {
+            return;
+        }
+        auto tid = tinfo.tid();
```
Comment (aceforeverd) [inline src/sdk/sql_cluster_router.cc:3283]: check `OfflineTableInfo`

File: src/cmd/sql_cmd_test.cc (hunk 2 above)
Comment (aceforeverd) [inline src/cmd/sql_cmd_test.cc]: disk_data_size update every 10min

File: src/sdk/db_sdk.cc [hunk 6]
```diff
@@ -284,9 +284,18 @@ std::shared_ptr<::openmldb::nameserver::TableInfo> DBSDK::GetTableInfo(const std
     return table_info;
 }
 
-std::vector<std::shared_ptr<::openmldb::nameserver::TableInfo>> DBSDK::GetTables(const std::string& db) {
+std::vector<std::shared_ptr<::openmldb::nameserver::TableInfo const>> DBSDK::GetTables(const std::string& db) {
     std::lock_guard<::openmldb::base::SpinMutex> lock(mu_);
-    std::vector<std::shared_ptr<::openmldb::nameserver::TableInfo>> tables;
+    std::vector<std::shared_ptr<::openmldb::nameserver::TableInfo const>> tables;
```
Comment (aceforeverd) [inline src/sdk/db_sdk.cc]: revert

File: src/cmd/sql_cmd_test.cc (hunk 2 above)
Comment (aceforeverd) [reply src/cmd/sql_cmd_test.cc]: unit: bytes

**Review by vagetablechicken — APPROVED** [review]

**Review by zhanghaohit — APPROVED** [review]

**Pull request merged**